pkg/lib/scoped/token_retriever.go

```diff
@@ -36,8 +37,15 @@ func (r *BearerTokenRetriever) Retrieve(reference *corev1.ObjectReference) (toke
 	}
 
 	if secret == nil {
-		err = fmt.Errorf("the service account does not have any API secret sa=%s/%s", sa.GetNamespace(), sa.GetName())
-		return
+		token, err = requestSAToken(r.kubeclient, sa)
+		if err != nil {
+			err = fmt.Errorf("creating service account token from TokenRequest API for sa=%s/%s; %v",
+				sa.GetNamespace(),
+				sa.GetName(),
+				err,
+			)
+			return
+		}
```

Comment:
Doesn't the token get immediately overridden by the line below? Should the `return` be on line 48?

Reply:
Good catch. I moved the `return` down a line which will ensure the token and error are returned before evaluating subsequent lines.